qa/integration-tests/src/test/java/io/camunda/zeebe/broker/it/clustering/SnapshotOnFollowerTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.it.clustering;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.broker.system.configuration.DataCfg;
+import io.camunda.zeebe.broker.system.configuration.ExporterCfg;
+import io.camunda.zeebe.exporter.api.Exporter;
+import io.camunda.zeebe.exporter.api.context.Context;
+import io.camunda.zeebe.exporter.api.context.Context.RecordFilter;
+import io.camunda.zeebe.exporter.api.context.Controller;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+
+public class SnapshotOnFollowerTest {
+
+  @Rule
+  public final ClusteringRule clusteringRule = new ClusteringRule(1, 2, 2, this::configureBroker);
+
+  @After
+  public void cleanUp() {
+    ControllableExporter.updatePosition(true);
+    ControllableExporter.EXPORTED_RECORDS.set(0);
+    ControllableExporter.RECORD_TYPE_FILTER.set(r -> true);
+    ControllableExporter.VALUE_TYPE_FILTER.set(r -> true);
+  }
+
+  @Test
+  public void shouldIncludeExportedPositionInSnapshotOnFollower() {
+    // given
+    ControllableExporter.updatePosition(true);
+    final var leader = clusteringRule.getLeaderForPartition(1);
+    final var follower = clusteringRule.getOtherBrokerObjects(leader.getNodeId()).get(0);
+
+    publishMessages();
+    ControllableExporter.updatePosition(false);
+    final var exportedPosition = ControllableExporter.lastUpdatedPosition;
```

Comment:
Not 100% sure but I have the feeling that this has potential for a flaky test ? :thinking:

Reply:
Like it is already in the exporter update position branch and we read it before we write the other value. I think two properties are hard to synchronize, even with volatile. Either we synchronize via a thread, which is the only one who has access to the properties or we use the synchronize key word or other locks. :shrug:

Reply:
Ya. There is a chance for flakyness. It would be correct if we move reading the lastUpdatedPosition into the awaitility loop. So eventually it will read the correct value.